### DISCUSSION TITLE
Session cookie expiration

### DISCUSSION THREAD

**Discussion by CountParadox** [bitbybyte/fantiadl#50] [Q&A] [answered]
Hey!

Thanks for the scripts, I have added this to one of my servers and it runs hourly with Crontab.
I had to specify python3 because python kept complaining that it didn't have bs4, even though its installed.

Its now running fine and downloading my subscriptions. Any ideas how long the token is valid for?

**Comment by bitbybyte** [comment, marked as answer]
In my experience, the session cookie doesn't actually change, but it may expire every few months. If you were scheduling it, I would say every month is a very safe cushion to login again to revalidate the cookie.

**Reply by CountParadox** [reply]
Oh cool, a month is better than I thought! I was worried it will be a few days. 

Any ideas if I can script getting the key from command line somehow? 🤔

**Reply by bitbybyte** [reply]
Logging in requires a CAPTCHA token (see https://github.com/bitbybyte/fantiadl/issues/28), so while you could retrieve a session cookie programatically, you will still need to login from the browser to validate the session if it ever gets invalidated. I would have scripted it myself if I could. 🙂
